doc/docs/Python_Tutorials/Mode_Decomposition.md

```diff
@@ -299,6 +300,184 @@ The diffraction orders/modes are a finite set of propagating planewaves. The wav
+sim.load_minus_flux_data(refl_flux,input_flux_data)
+
+tran_pt = mp.Vector3(0.5*sx-dpml-0.5*dpad,0,0)
+tran_flux = sim.add_flux(fcen, 0, 1, mp.FluxRegion(center=tran_pt, size=mp.Vector3(0,sy,0)))
+
+sim.run(until_after_sources=400)
+
+nm_r = np.floor((fcen*ng-k.y)*gp)-np.ceil((-fcen*ng-k.y)*gp) # number of reflected orders
+if theta_in == 0:
+  nm_r = nm_r/2 # since eig_parity removes degeneracy in y-direction
+nm_r = int(nm_r)
+  
+res = sim.get_eigenmode_coefficients(refl_flux, range(1,nm_r+1), eig_parity=eig_parity)
+r_coeffs = res.alpha
+
+Rsum = 0
+for nm in range(nm_r):
+  r_kdom = res.kdom[nm]
+  Rmode = abs(r_coeffs[nm,0,1])**2/input_flux[0]
+  r_angle = np.sign(r_kdom.y)*math.acos(r_kdom.x/(ng*fcen))
+  print("refl:, {}, {:.2f}, {:.8f}".format(nm,math.degrees(r_angle),Rmode))
+  Rsum += Rmode
+    
+nm_t = np.floor((fcen-k.y)*gp)-np.ceil((-fcen-k.y)*gp)       # number of transmitted orders
+if theta_in == 0:
+  nm_t = nm_t/2 # since eig_parity removes degeneracy in y-direction
+nm_t = int(nm_t)
+
+res = sim.get_eigenmode_coefficients(tran_flux, range(1,nm_t+1), eig_parity=eig_parity)
+t_coeffs = res.alpha
+
+Tsum = 0
+for nm in range(nm_t):
+  t_kdom = res.kdom[nm]
+  Tmode = abs(t_coeffs[nm,0,0])**2/input_flux[0]
+  t_angle = np.sign(t_kdom.y)*math.acos(t_kdom.x/fcen)
+  print("tran:, {}, {:.2f}, {:.8f}".format(nm,math.degrees(t_angle),Tmode))
+  Tsum += Tmode
+
+print("mode-coeff:, {:.6f}, {:.6f}, {:.6f}".format(Rsum,Tsum,Rsum+Tsum))
+
+r_flux = mp.get_fluxes(refl_flux)
+t_flux = mp.get_fluxes(tran_flux)
+Rflux = -r_flux[0]/input_flux[0]
+Tflux =  t_flux[0]/input_flux[0]
+print("poynting-flux:, {:.6f}, {:.6f}, {:.6f}".format(Rflux,Tflux,Rflux+Tflux))
+```
+
+Results are computed for a single wavelength of 0.5 μm. The pulsed planewave is incident at an angle of 10.7° (`theta_in`). Its spatial profile is defined using the source amplitude function (`pw_amp`). This function takes two arguments, the wavevector and a point in space (both `mp.Vector3`s), and returns a function of one argument which returns the planewave amplitude at that point. Two modifications to the original simulation are necessary for mitigating the intrinsic discretization effects of the Yee grid for oblique planewaves: (1) the pulse bandwidth is narrowed and (2) the anisotropic `PML` is replaced with an isotropic `Absorber`. Also, the `stop_when_fields_decayed` termination criteria is replaced with `until_after_sources`. As a general rule of thumb, the more oblique the planewave source, the longer the run time required to ensure accurate results. Note that there is a second line monitor between the source and the grating for computing the reflectance. The angle of each reflected/transmitted mode, which can be positive or negative, is computed using its dominant planewave vector. Since the oblique source breaks the symmetry in the $y$ direction, each diffracted order must be computed separately. In total, there are 59 reflected and 39 transmitted orders.
```

Comment:
Ardavan, you wrote that you narrowed the bandwidth to "mitigate discretization effects".  What are you referring to here?  The discretization error at a given frequency should be independent of the bandwidth, by linearity.

One of my students pointed out the corresponding sentence in the current manual (https://meep.readthedocs.io/en/latest/Python_Tutorials/Mode_Decomposition/#reflectance-and-transmittance-spectra-for-planewave-at-oblique-incidence) which traces back to this PR, and it is rather confusing to me.

Reply:
(Also it's not clear why you need to replace the PML with a scalar absorber — PML seems like it should work fine here?)

Reply:
I was referring here to the numerical dispersion which arises for planewaves propagating at an oblique angle relative to the Cartesian Yee grid. When I was setting up this tutorial, I found that the broader the bandwidth of the pulsed source, the longer it was taking for the fields to decay away and thus for the simulation to be terminated. I think there could be some interaction of the oblique waves and the absorbing boundary layers which was causing poor attenuation. The workaround was to define a narrow-bandwidth source.

I discovered after this PR that the choice of `Absorber` and `PML` actually did not affect the results. This is why the `Absorber` was replaced with a `PML` in a [subsequent change](https://github.com/NanoComp/meep/pull/605/files#diff-0205fe8da836afa86c9d910a14fb9eab8d0fc270c086bd8db2beb087b11a7207R21) in #605.

Reply:
This is not a discretization issue — if the bandwidth is too broad, then you are creating glancing-angle waves, which aren't absorbed well by the PML (or by an absorber).